src/comm/UdpIODevice.cc

```diff
@@ -0,0 +1,42 @@
+/****************************************************************************
+ *
+ *   (c) 2009-2018 QGROUNDCONTROL PROJECT <http://www.qgroundcontrol.org>
+ *
+ * QGroundControl is licensed according to the terms in the file
+ * COPYING.md in the root of the source code directory.
+ *
+ ****************************************************************************/
+
+#include "UdpIODevice.h"
+#include <algorithm>
+
+UdpIODevice::UdpIODevice(QObject *parent) : QUdpSocket(parent)
+{
+}
+
+bool UdpIODevice::canReadLine() const
+{
+    return _buffer.indexOf('\n') > -1 || hasPendingDatagrams();
```

Comment:
The pending datagrams do not necessarily contain a newline.

You should read any pending datagrams in, then

`return _buffer.indexOf('\n') > -1`

Reply:
Maybe use this class should use the parent readyRead signal to read in the datagrams as soon as they are ready. http://doc.qt.io/qt-5/qiodevice.html#readyRead